test/unit/base_test.rb

```diff
@@ -202,5 +202,39 @@ class Aggregate::BaseTest < ActiveSupport::TestCase
         @instance.name
       end
     end
+
+    context "secret keys" do
+      setup do
+        Aggregate.reset
+      end
+      should "return nil when keys_list is nil" do
+        assert_equal [], Aggregate::Base.secret_keys_from_config
+      end
+
+      should "return nil when keys_list is something other than a string or hash" do
+        Aggregate.configure do |config|
+          config.keys_list = Object.new
+        end
+        assert_equal [], Aggregate::Base.secret_keys_from_config
+      end
+
+      should "return a array when keys_list is a string" do
+        Aggregate.configure do |config|
+          config.keys_list = "eAtJqKqF2IP0+djGMGq55Hk0vW017M+SEZISldB7Ofw="
+        end
+
+        assert Aggregate.configuration.keys_list.is_a?(String)
+        assert Aggregate::Base.secret_keys_from_config.is_a?(Array)
```

Comment:
I agree that type checking is smelly - but I think there are cases where you want to type check if you explicitly only want specific types (I pass in a string, and expect an array back) 

Will fix to assert_true